src/main/java/org/aya/intellij/parser/AyaParserDefinition.java

```diff
@@ -5,35 +5,25 @@
 import com.intellij.lang.PsiParser;
 import com.intellij.lexer.FlexAdapter;
 import com.intellij.lexer.Lexer;
-import com.intellij.openapi.fileTypes.SyntaxHighlighter;
-import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory;
 import com.intellij.openapi.project.Project;
-import com.intellij.openapi.vfs.VirtualFile;
-import com.intellij.psi.FileViewProvider;
-import com.intellij.psi.PsiElement;
-import com.intellij.psi.PsiFile;
+import com.intellij.psi.builder.FleetPsiBuilder;
 import com.intellij.psi.tree.IFileElementType;
 import com.intellij.psi.tree.TokenSet;
 import org.aya.intellij.AyaLanguage;
-import org.aya.intellij.actions.SyntaxHighlight;
-import org.aya.intellij.psi.AyaPsiFile;
 import org.aya.intellij.psi.types.AyaPsiElementTypes;
 import org.aya.intellij.psi.types.AyaPsiTokenType;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
 
-public class AyaParserDefinition extends SyntaxHighlighterFactory implements ParserDefinition {
-  private final @NotNull IFileElementType FILE = new IFileElementType(AyaLanguage.INSTANCE);
+public class AyaParserDefinition implements ParserDefinition {
+  private final @NotNull IFileElementType FILE = new IFileElementType(AyaLanguage.INSTANCE) {
+    @Override public void parse(@NotNull FleetPsiBuilder<?> builder) {
+    }
+  };
 
   public static @NotNull Lexer createLexer() {
     return new FlexAdapter(new _AyaPsiLexer());
   }
 
-  @Override
-  public @NotNull SyntaxHighlighter getSyntaxHighlighter(@Nullable Project project, @Nullable VirtualFile virtualFile) {
-    return new SyntaxHighlight();
-  }
-
```

Comment:
Then there should be a new `SyntaxHighlighter` in plugin.xml

Reply:
:sob: